build/scripts/generate-compiler-code.ps1

```diff
@@ -0,0 +1,114 @@
+[CmdletBinding(PositionalBinding=$false)]
+param ([switch]$test = $false)
+
+Set-StrictMode -version 2.0
+$ErrorActionPreference="Stop"
+
+function Run-Tool($tool, $toolArgs) {
+    $toolName = Split-path -leaf $tool
+    Write-Host "Running $toolName"
+    if (-not (Test-Path $tool)) {
+        throw "$tool does not exist"
+    }
+
+    Invoke-Expression "& `"$coreRun`" `"$tool`" $toolArgs"
+    if ((-not $?) -or ($lastexitcode -ne 0)) {
+        throw "Failed"
+    }
+}
+
+function Run-LanguageCore($language, $languageSuffix, $languageDir, $syntaxTool, $errorFactsTool, $generatedDir, $generatedTestDir) {
+    $syntaxFilePath = Join-Path $languageDir "Syntax\Syntax.xml"
+    $syntaxTestFilePath = Join-Path $generatedTestDir "Syntax.Test.xml.Generated.$($languageSuffix)"
+    $boundFilePath = Join-Path $languageDir "BoundTree\BoundNodes.xml"
+    $boundGeneratedFilePath = Join-Path $generatedDir "BoundNodes.xml.Generated.$($languageSuffix)"
+    $errorFileName = if ($language -eq "CSharp") { "ErrorCode.cs" } else { "Errors.vb" }
+    $errorFilePath = Join-Path $languageDir "Errors\$errorFileName"
+    $errorGeneratedFilePath = Join-Path $generatedDir "ErrorFacts.Generated.$($languageSuffix)"
+
+    Create-Directory $generatedDir
+    Create-Directory $generatedTestDir
+    Run-Tool $syntaxTool "`"$syntaxFilePath`" `"$generatedDir`""
+    Run-Tool $syntaxTool "`"$syntaxFilePath`" `"$syntaxTestFilePath`" /test"
+    Run-Tool $boundTreeGenerator "$language `"$boundFilePath`" `"$boundGeneratedFilePath`""
+    Run-Tool $errorFactsTool "`"$errorFilePath`" `"$errorGeneratedFilePath`""
+}
+
+# Test the contents of our generated files to ensure they are equal.  Compares our checked
+# in code with the freshly generated code. 
+function Test-GeneratedContent($generatedDir, $scratchDir) {
+    $algo = "MD5"
+    foreach ($fileName in (Get-ChildItem $scratchDir)) { 
+        Write-Host "Checking $fileName"
+        $realFilePath = Join-Path $generatedDir $fileName
+        $scratchFilePath = Join-Path $scratchDir $fileName
+        $scratchHash = (Get-FileHash $scratchFilePath -algorithm $algo).Hash
+        $realHash = (Get-FileHash $realFilePath -algorithm $algo).Hash
+        if ($scratchHash -ne $realHash) { 
+            Write-Host "Files are out of date"
+            Write-Host "Run $(Join-Path $PSScriptRoot generate-compiler-code.ps1) to refresh"
+            throw "Files are out of date"
```

Comment:
"Out of sync with the xml definition" ?
Perhaps also include the file name in the error.